TestProjects/ShaderGraph/Assets/CommonAssets/Editor/ImportUpdateTests.cs

```diff
@@ -133,6 +134,36 @@ public void TestImportAsset(string unityLocalPath, string fullPath)
                 string shader2 = generator2.generatedShader;
 
                 Assert.AreEqual(shader, shader2, $"Importing the graph {unityLocalPath} twice resulted in different generated shaders.");
+
+                // Texture test won't work on platforms that don't support more than 16 samplers
+
+                bool isGL =
+                    (SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLCore) ||
+                    (SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLES2) ||
+                    (SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLES3);
+
+                bool isOSX =
+                    (Application.platform == RuntimePlatform.OSXEditor) ||
+                    (Application.platform == RuntimePlatform.OSXPlayer);
+
```

Comment:
Vulkan doesn't guarantee more than 16 either.  I suspect most desktop implementations actually provide more than 16, but the hardware requirements for the spec are derived from OpenGL ES 3.1, so it's not guaranteed without checking.

Reply:
Yeah, it at least passes on Windows Editor / Vulkan on my box.